runner/runner/main.py

```diff
@@ -252,85 +260,104 @@ def process_plugin(plugin_config: Mapping[str, Any]) -> Path:
+                        *illixr_cmd_list,
+                    ],
+                    ("$cmd",): illixr_cmd_list,
+                    ("$quoted_cmd",): [shlex.quote(shlex.join(illixr_cmd_list))],
+                    ("$env",): env_list,
+                },
+            )
         )
-
-    if not openxr_app_bin_path.exists():
-        raise RuntimeError(f"{action_name} Failed to build openxr_app (mainline={is_mainline}, path={openxr_app_bin_path})")
-
-    if is_mainline:
-        monado_target_name : str  = "monado-service"
-        monado_target_dir  : Path = monado_path / "build" / "src" / "xrt" / "targets" / "service"
-        monado_target_path : Path = monado_target_dir / monado_target_name
-
-        if not monado_target_path.exists():
-            raise RuntimeError(f"[{action_name}] Failed to build monado (mainline={is_mainline}, path={monado_target_path})")
-
-        env_monado_service: Mapping[str, str] = dict(**os.environ, **env_monado, **env_gpu)
-
-        ## Open the Monado service application in the background
-        monado_service_proc = subprocess.Popen([str(monado_target_path)], env=env_monado_service, stdin=PIPE, stdout=PIPE, stderr=PIPE)
-
-    ## Give the Monado service some time to boot up and the user some time to initialize VIO
-    time.sleep(5)
-
-    subprocess_run(
-        [str(openxr_app_bin_path)],
-        env_override=dict(
-            ILLIXR_DEMO_DATA=str(demo_data_path),
-            ILLIXR_OFFLOAD_ENABLE=str(enable_offload_flag),
-            ILLIXR_ALIGNMENT_ENABLE=str(enable_alignment_flag),
-            ILLIXR_ENABLE_VERBOSE_ERRORS=str(config["enable_verbose_errors"]),
-            ILLIXR_ENABLE_PRE_SLEEP=str(config["enable_pre_sleep"]),
-            KIMERA_ROOT=config["action"]["kimera_path"],
-            AUDIO_ROOT=config["action"]["audio_path"],
-            REALSENSE_CAM=str(realsense_cam_string),
-            **env_monado,
-            **env_gpu,
-        ),
-        check=True,
     )
 
-    if is_mainline:
-        ## Close and clean up the Monado service application
-        try:
-            outs, errs = monado_service_proc.communicate(timeout=1)
-        except subprocess.TimeoutExpired:
-            monado_service_proc.kill()
-            outs, errs = monado_service_proc.communicate()
+    ## Launch the Monado service before any OpenXR apps are opened
+    monado_service_proc = subprocess.Popen(actual_cmd_list, env=env_override)
 
-            ## Clean up leftover socket. It can only either be in $XDG_RUNTIME_DIR or /tmp
-            Path(env_monado_service['XDG_RUNTIME_DIR'] + "/monado_comp_ipc").unlink(missing_ok=True)
-            Path("/tmp/monado_comp_ipc").unlink(missing_ok=True)
-
-        print("\nstdout:\n")
-        sys.stdout.buffer.write(outs)
+    ## Give the Monado service some time to boot up and the user some time to initialize VIO
+    time.sleep(10)
+
+    # Launch all OpenXR apps after the service is launched
+    for openxr_app_bin_path in openxr_app_bin_paths:
```

Comment:
Should we launch all the apps?

Reply:
I think Jeffrey added this to support multi-client. Hasn't changed this part from this script.

Reply:
Yes, this also addresses the previous comment because all potentially compilable OpenXR apps (like openxr-simple-example) will be built here now. It's a bit rough around the edges but I think it's okay to keep for now, unless we don't want to support multiple clients yet.